src/FixedDataTableCell.js

```diff
@@ -233,62 +209,50 @@ class FixedDataTableCell extends React.Component {
         'public/fixedDataTableCell/alignRight': props.align === 'right',
         'public/fixedDataTableCell/highlighted': props.highlighted,
         'public/fixedDataTableCell/main': true,
-        'public/fixedDataTableCell/hasReorderHandle': !!props.onColumnReorder,
-        'public/fixedDataTableCell/reordering': this.state.isReorderingThisColumn,
       }),
       props.className,
     );
 
-    var columnResizerComponent;
-    if (props.onColumnResize) {
-      var columnResizerStyle = {
-        height
-      };
-      columnResizerComponent = (
-        <div
-          className={cx('fixedDataTableCellLayout/columnResizerContainer')}
-          style={columnResizerStyle}
-          onMouseDown={this._onColumnResizerMouseDown}
-          onTouchStart={this.props.touchEnabled ? this._onColumnResizerMouseDown : null}
-          onTouchEnd={this.props.touchEnabled ? this._suppressEvent : null}
-          onTouchMove={this.props.touchEnabled ? this._suppressEvent : null}>
-          <div
-            className={joinClasses(
-              cx('fixedDataTableCellLayout/columnResizerKnob'),
-              cx('public/fixedDataTableCell/columnResizerKnob'),
-            )}
-            style={columnResizerStyle}
-          />
-        </div>
-      );
-    }
-
-    var columnReorderComponent;
-    if (props.onColumnReorder) { //header row
-      columnReorderComponent = (
-        <FixedDataTableColumnReorderHandle
-          columnKey={this.columnKey}
-          touchEnabled={this.props.touchEnabled}
-          onMouseDown={this._onColumnReorderMouseDown}
-          onTouchStart={this._onColumnReorderMouseDown}
-          height={height}
-          {...this.props}
-        />
-      );
-    }
-
     var cellProps = {
       columnKey,
       height,
-      width
+      width,
     };
+    // Todo(deshpsuy): Clean the unused props
+    if (this.props.isHeader) {
+      cellProps = {
+        ...cellProps,
+        tableHeight: this.props.tableHeight,
+        left: this.props.left,
+        cellGroupLeft: this.props.cellGroupLeft,
+        touchEnabled: this.props.touchEnabled,
+        isRTL: this.props.isRTL,
+        scrollX: this.props.scrollX,
+        isFixed: this.props.isFixed,
+        availableScrollWidth: this.props.availableScrollWidth,
+        maxScrollX: this.props.maxScrollX,
+        cellGroupColumnWidths: this.props.cellGroupColumnWidths,
+        scrollToX: this.props.scrollToX,
+        getCellGroupWidth: this.props.getCellGroupWidth,
+        columnGroupWidth: this.props.columnGroupWidth
+      };
+    }
 
     if (props.rowIndex >= 0) {
       cellProps.rowIndex = props.rowIndex;
     }
 
     var content;
-    if (React.isValidElement(props.cell)) {
+    if (this.props.isHeader && (this.props.onColumnResizeEndCallback || this.props.onColumnReorderEndCallback)) {
```

Comment:
Can you put a comment here explaining the code path under this `if` condition is only used for maintaining backward compatibility for the older resize/reoder API ?

Reply:
Added